### PR TITLE
PWGEM/PhotonMeson: update PCM for MC

### PR DESCRIPTION
**Pull request by dsekihat**
This PR contains following commits.
PWGEM/PhotonMeson: update MC tasks
PWGEM/PhotonMeson: update createPCM with track-collision associator

### CODE REVIEW

File: PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx [hunk 1]
```diff
@@ -110,22 +111,22 @@ struct createEMReducedMCEvent {
...
-          && (abs(pdg) != 223 || !mctrack.producedByGenerator()) // omega(782)
-          && (abs(pdg) != 331 || !mctrack.producedByGenerator()) // eta'(958)
-          && (abs(pdg) != 333 || !mctrack.producedByGenerator()) // phi(1020)
+          && (abs(pdg) != 111 || !IsPhysicalPrimary(mctrack, mcTracks)) // pi0
+          && (abs(pdg) != 113 || !IsPhysicalPrimary(mctrack, mcTracks)) // rho(770)
+          && (abs(pdg) != 211 || !IsPhysicalPrimary(mctrack, mcTracks)) // changed pion
+          && (abs(pdg) != 221 || !IsPhysicalPrimary(mctrack, mcTracks)) // eta
+          && (abs(pdg) != 223 || !IsPhysicalPrimary(mctrack, mcTracks)) // omega(782)
+          && (abs(pdg) != 331 || !IsPhysicalPrimary(mctrack, mcTracks)) // eta'(958)
+          && (abs(pdg) != 333 || !IsPhysicalPrimary(mctrack, mcTracks)) // phi(1020)
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx:129]: the check` || !IsPhysicalPrimary(mctrack, mcTracks))` probably can be changed to only check this once right?

File: PWGEM/PhotonMeson/TableProducer/createPCM.cxx [hunk 2]
```diff
@@ -239,20 +240,58 @@ struct createPCM {
...
 
       for (auto& [ele, pos] : combinations(CombinationsFullIndexPolicy(negTracks_coll, posTracks_coll))) {
         fillV0Table(collision, ele, pos);
       }
     } // end of collision loop
   }   // end of process
+  PROCESS_SWITCH(createPCM, processSA, "create V0s with stand-alone way", true);
+
+  Preslice<aod::TrackAssoc> trackIndicesPerCollision = aod::track_association::collisionId;
+  void processTrkCollAsso(aod::TrackAssoc const& trackIndices, FullTracksExt const& tracks, aod::Collisions const& collisions, aod::BCsWithTimestamps const&)
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/TableProducer/createPCM.cxx:252]: is this process function not looping over anything automatically? Just to be sure, there is no double loop over aod::TrackAssoc ^^"

Comment (dsekihat) [reply PWGEM/PhotonMeson/TableProducer/createPCM.cxx:252]: no.

Comment (mhemmer-cern) [reply PWGEM/PhotonMeson/TableProducer/createPCM.cxx:252]: okay good!

**Review by mhemmer-cern — APPROVED** [review]
I think this looks okay, minor comments I had can also be resolved later in another PR.
@mikesas @rbailhac

File: PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx (hunk 1 above)
Comment (dsekihat) [reply PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx:129]: probably, yes. I see your point. I will fix this in next commits.

**Review by rbailhac — APPROVED** [review]

**Pull request merged**